### PR TITLE
[Consensus] Support Node 20

### PR DESCRIPTION
**Pull request by Rishi556**
Adds support for Node.JS 20. 

The error message logged during deployment needed to be changed from using node's error message to a custom error message as node's internal changed how the message would be spit out. For now I put the deployment date as after hive block `76149072` which is approximately July 25th.

Old message from node: `SyntaxError: Unexpected identifier`
New message from node: `SyntaxError: Unexpected identifier 'CODE'`

To handle older blocks, anything with "SyntaxError: Unexpected identifier" forcibly becomes just "SyntaxError: Unexpected identifier", removing the extra stuff which stated what was the unidentified variable.

### CODE REVIEW

**Comment by bt-cryptomancer** [comment]
@Rishi556 I've converted this to draft status as I'm not comfortable making such a fundamental change for the time being. We can revisit this in September when I'm back from summer travels.

**Comment by Rishi556** [comment]
I do think we should try and update as new releases are made, especially since Node now has a team dedicated to performance. Doing updates in smaller batches and keeping up will make it a lot easier than trying to fix support for a few node versions at once which might have many breaking changes, vs doing 1 at a time might lead to just a few. BTW this will need to be changed before deployment to change the block num.

**Comment by Rishi556** [comment]
@bt-cryptomancer Can you take a look at this. We'll have to change block number but I'd like to get this one merged in before Node 22 gets released in about 2 months

File: libs/SmartContracts.js [hunk 1]
```diff
@@ -244,8 +244,14 @@ class SmartContracts {
       }
       return { logs: { errors: ['parameters name and code are mandatory and they must be strings'] } };
     } catch (e) {
-      // console.error('ERROR DURING CONTRACT DEPLOYMENT: ', name, e);
-      return { logs: { errors: [`${e.name}: ${e.message}`] } };
+      // log.error('ERROR DURING CONTRACT DEPLOYMENT: ', e);
```
Comment (eonwarped) [inline libs/SmartContracts.js:247]: let's re-enable this log

File: libs/SmartContracts.js [hunk 2]
```diff
@@ -244,8 +244,14 @@ class SmartContracts {
...
+      if (refHiveBlockNumber <= 76149072) { // Approximately Sunday July 25 UTC
```
Comment (eonwarped) [inline libs/SmartContracts.js:248]: just tracking needed change to block number

**Review by eonwarped — CHANGES_REQUESTED** [review]
Small comment, looks fine to me.

File: libs/SmartContracts.js (hunk 2 above)
Comment (Rishi556) [reply libs/SmartContracts.js:248]: Maintainers should have the ability to make this change themselves once they are ready to merge this in.

File: libs/SmartContracts.js (hunk 1 above)
Comment (Rishi556) [reply libs/SmartContracts.js]: Resolved in https://github.com/hive-engine/hivesmartcontracts/pull/46/commits/77e64027166eddf5c4424684c44c6b4e66768064

**Comment by Rishi556** [comment]
Node 16 is EOL as well, thoughts on dropping it from the test matrix?

File: libs/SmartContracts.js (hunk 2 above)
Comment (eonwarped) [reply libs/SmartContracts.js:248]: Good point, will make sure to figure out when the next window we would like to make the next change is.

**Comment by eonwarped** [comment]
> Node 16 is EOL(and is approaching 6 months of EOL) as well, thoughts on dropping it from the test matrix?

deferring to @bt-cryptomancer

**Comment by bt-cryptomancer** [comment]
@Rishi556 I'm fine to have Node 16 removed from the test matrix.  If we're going to do that, let's also update the minimum node version in the package.json to 18+

I will update the trigger block number after merging this to QA and after other consensus breaking changes are merged in.

**Comment by bt-cryptomancer** [comment]
Looks good now, but conflict needs to be resolved before merging.

**Pull request merged**